lib/chef/provider/windows_task.rb

```diff
@@ -0,0 +1,396 @@
+            Chef::Log.info "#{@new_resource} task ran"
+          end
+        else
+          Chef::Log.warn "#{@new_resource} task doesn't exists - nothing to do"
+        end
+      end
+
+      def action_delete
+        if @current_resource.exists
+          # always need to force deletion
+          run_schtasks 'DELETE', 'F' => ''
+          new_resource.updated_by_last_action true
+          Chef::Log.info "#{@new_resource} task deleted"
+        else
+          Chef::Log.warn "#{@new_resource} task doesn't exists - nothing to do"
+        end
+      end
+
+      def action_end
+        if @current_resource.exists
+          if @current_resource.status != :running
+            Chef::Log.debug "#{@new_resource} is not running - nothing to do"
+          else
+            run_schtasks 'END'
+            @new_resource.updated_by_last_action true
+            Chef::Log.info "#{@new_resource} task ended"
+          end
+        else
+          Chef::Log.warn "#{@new_resource} task doesn't exist - nothing to do"
+        end
+      end
+
+      def action_enable
+        if @current_resource.exists
+          if @current_resource.enabled
+            Chef::Log.debug "#{@new_resource} already enabled - nothing to do"
+          else
+            run_schtasks 'CHANGE', 'ENABLE' => ''
+            @new_resource.updated_by_last_action true
+            Chef::Log.info "#{@new_resource} task enabled"
+          end
+        else
+          Chef::Log.fatal "#{@new_resource} task doesn't exist - nothing to do"
+          raise Errno::ENOENT, "#{@new_resource}: task does not exist, cannot enable"
+        end
+      end
+
+      def action_disable
+        if @current_resource.exists
+          if @current_resource.enabled
+            run_schtasks 'CHANGE', 'DISABLE' => ''
+            @new_resource.updated_by_last_action true
+            Chef::Log.info "#{@new_resource} task disabled"
+          else
+            Chef::Log.warn "#{@new_resource} already disabled - nothing to do"
+          end
+        else
+          Chef::Log.warn "#{@new_resource} task doesn't exist - nothing to do"
+        end
+      end
+
+      private
+
+      # rubocop:disable Style/StringLiteralsInInterpolation
+      def run_schtasks(task_action, options = {})
+        cmd = "schtasks /#{task_action} /TN \"#{@new_resource.task_name}\" "
+        options.keys.each do |option|
+          cmd += "/#{option} "
+          cmd += "\"#{options[option].to_s.gsub('"', "\\\"")}\" " unless options[option] == ''
+        end
+        Chef::Log.debug('running: ')
+        Chef::Log.debug("    #{cmd}")
+        shell_out!(cmd, returns: [0])
+      end
+      # rubocop:enable Style/StringLiteralsInInterpolation
+
+      def task_need_update?
+        # gsub needed as schtasks converts single quotes to double quotes on creation
+        @current_resource.command != @new_resource.command.tr("'", '"') ||
+        @current_resource.user != @new_resource.user ||

```

Comment:
If we run `windows_task` on non-English system, for example, Russian version of Windows, the task will be created with RunAsUser: `СИСТЕМА`. As a result task will be created on each run, as a
```ruby
@current_resource.user != @new_resource.user
```
will never match. I think, it's because encoding issues. Maybe we should use SID, instead of hardcoded system names.